Ullaakut/FactoriHub
# Fix copy blueprint button and integrate homepage

## Goal of this PR

This PR is part of #2 and does the following:

* Fixes the homepage to properly get Author data from entries (using `Eager` on the database handle)
* Makes `1.0.0` the hardcoded latest since for now real Factorio versioning is not fetched automatically
* Hardcodes a user ID for now (needs to be created manually in the DB for testing, but will not break the app if no user is found)
	* To reproduce it, run the following commands:
	* `buffalo dev`
	* `psql postgres://postgres:postgres@127.0.0.1:5432/factorihub_development\?sslmode=disable`
	* `INSERT INTO public.users (id, name, email, created_at, updated_at) VALUES ('c756b686-f97d-11ea-adc1-0242ac120002', 'JohnDoe', 'john.doe@test.com', '2004-10-19 10:23:54+02', '2004-10-19 10:23:54+02');`
	* Any entry created should now have `JohnDoe` as its user both in the homepage and individual blueprint details page
* Fixes the copy blueprint string button
	* The previous version worked on the individual blueprint page but could not work on the homepage due to a Buffalo issue
	* The new version also supports having multiple different blueprint strings on a single page and a button for each
* Integrates the mockup for the home page
	* The updates to the search bar and filters will be done in a separate PR

## How to test it

* `buffalo dev`
* Visit `localhost:3000`
* The home page should be empty by default because your database is fresh
* Create a new blueprint by clicking the `Upload` button and filling the information
* You should then be redirected to the homepage and your blueprint should now be there
* If it's a book and not a single blueprint, a label should be visible showing the amount of blueprints in the book
* The button at the top right of the card should allow you to copy the blueprint string
* Clicking anywhere else on the card should redirect to the blueprint details
